tensorflow_io/core/kernels/azfs_kernels.cc

```diff
@@ -356,249 +454,235 @@ class AzBlobWritableFile : public WritableFile {
-                 });
+  TF_SetStatus(status, TF_OK, "");
+}
 
-  std::copy_if(std::begin(blobs), std::end(blobs), std::back_inserter(*results),
-               [&pattern](const std::string &full_path) {
-                 return Env::Default()->MatchPath(full_path, pattern);
-               });
+static void NewWritableFile(const TF_Filesystem* filesystem, const char* path,
+                            TF_WritableFile* file, TF_Status* status) {
+  std::string account, container, object;
+  ParseAzBlobPath(path, false, &account, &container, &object, status);
+  if (TF_GetCode(status) != TF_OK) {
+    return;
+  }
+  file->plugin_file = new AzBlobWritableFile(account, container, object);
 
-  return Status::OK();
+  TF_SetStatus(status, TF_OK, "");
 }
 
-Status AzBlobFileSystem::DeleteFile(const std::string &fname) {
+static void NewAppendableFile(const TF_Filesystem* filesystem, const char* path,
+                              TF_WritableFile* file, TF_Status* status) {
   std::string account, container, object;
-  TF_RETURN_IF_ERROR(
-      ParseAzBlobPath(fname, false, &account, &container, &object));
-  auto blob_client = CreateAzBlobClientWrapper(account);
-
-  blob_client.delete_blob(container, object);
-  if (errno != 0) {
-    return errors::Internal("Failed to delete ", fname, " (", errno_to_string(),
-                            ")");
+  ParseAzBlobPath(path, false, &account, &container, &object, status);
+  if (TF_GetCode(status) != TF_OK) {
+    return;
   }
+  file->plugin_file = new AzBlobWritableFile(account, container, object);
 
-  return Status::OK();
+  TF_SetStatus(status, TF_OK, "");
+}
+
+static void NewReadOnlyMemoryRegionFromFile(const TF_Filesystem* filesystem,
+                                            const char* path,
+                                            TF_ReadOnlyMemoryRegion* region,
+                                            TF_Status* status) {
+  TF_SetStatus(status, TF_UNIMPLEMENTED,
+               "NewReadOnlyMemoryRegionFromFile not implemented");
 }
 
-Status AzBlobFileSystem::CreateDir(const std::string &dirname) {
+static void CreateDir(const TF_Filesystem* filesystem, const char* path,
+                      TF_Status* status) {
   std::string account, container, object;
-  TF_RETURN_IF_ERROR(
-      ParseAzBlobPath(dirname, true, &account, &container, &object));
+  ParseAzBlobPath(path, true, &account, &container, &object, status);
+  if (TF_GetCode(status) != TF_OK) {
+    return;
+  }
   if (container.empty()) {
-    return errors::FailedPrecondition("Cannot create storage accounts");
+    TF_SetStatus(status, TF_FAILED_PRECONDITION,
+                 "Cannot create storage accounts");
+    return;
   }
 
   // Blob storage has virtual folders. We can make sure the container exists
   auto blob_client_wrapper = CreateAzBlobClientWrapper(account);
 
   if (blob_client_wrapper.container_exists(container)) {
-    return Status::OK();
+    TF_SetStatus(status, TF_OK, "");
```

Comment:
Maybe add a message saying container exists?

Reply:
@kvignesh1420 In Azure, the container is in reference to S3's bucket and we treat them as part of the `directory` path, i.e., `/container/path/to/file` is considered as one path with AZFS like other file systems (not to treat `container` part differently). Given this function is to see if the `path` already exists. I think it might be fine to not single out container.